Sources/Realtime/Defaults.swift

```diff
@@ -228,3 +234,28 @@ public struct ChannelOptions {
     self.broadcastAcknowledge = broadcastAcknowledge
   }
 }
+
+/// Represents the different status of a push
+public enum PushStatus: RawRepresentable {
+    case ok
+    case error
+    case timeout
+    
+    public var rawValue: String {
+        switch self {
+        case .ok: return "ok"
+        case .error: return "error"
+        case .timeout: return "timeout"
+        }
+    }
+    
+    public init?(rawValue: String) {
+        switch rawValue {
+        case "ok": self = .ok
+        case "error": self = .error
+        case "timeout": self = .timeout
+        default: return nil
+        }
+    }
+    
+}
```

Comment:
You can rewrite this all simply as:

```suggestion
/// Represents the different status of a push
public enum PushStatus: String {
    case ok
    case error
    case timeout
}
```